server.test.js

```diff
@@ -3,196 +3,91 @@ let { server, game } = require('./server.js');
-                        .send({
-                          "act": "word",
-                          "word": "",
-                          "id": JSON.stringify(guesserID)
-                        });
-      expect(badWord.statusCode).toEqual(400);
-
-      const badid = await request(server)
-                        .post('/')
-                        .send({
-                          "act": "word",
-                          "word": "asd",
-                          "id": JSON.stringify(guesserID)
-                        });
-      expect(badid.statusCode).toEqual(401);
-      expect(game.wordToGuess).toEqual('debug');
-      done();
-    });
+  it('should connect on load', async (done) => {
+    expect(socketIsLoaded).toEqual(true)
+    done();
   });
 
-  describe('makeGuess', function() {
-
-    it('should be able to guess letters', async function(done) {
-      const res = await request(server)
-                        .post('/')
-                        .send({
-                          "act": "guess",
-                          "guess": "d",
-                          "id": JSON.stringify(guesserID)
-                        });
-      expect(res.statusCode).toEqual(200);
-      expect(res.body.display).toEqual(['d','_','_','_','_']);
-      expect(res.body.guesses).toEqual(6);
-      expect(res.body.isOver).toEqual(false);
-      expect(game.wordToGuess).toEqual('debug');
+  it('should be able to join a game', async (done) => {
+    callCount = 0;
+    socket.emit('joinGame', true);
+    setTimeout(() => {
+      expect(callCount).toEqual(1);
+      expect(result).toEqual({
+        display: [],
+        isOver: false,
+        remainingGuesses: 6,
+        attempts: [],
+        isGameReady: false
+      })
       done();
-    });
-
-    it('should return an error for malformed bodies', async function(done) {
-      const badWord = await request(server)
-                        .post('/')
-                        .send({
-                          "act": "guess",
-                          "guess": "",
-                          "id": JSON.stringify(guesserID)
-                        });
-      expect(badWord.statusCode).toEqual(400);
+    }, 50)
+  });
 
-      const badid = await request(server)
-                        .post('/')
-                        .send({
-                          "act": "guess",
-                          "guess": "a",
-                          "id": JSON.stringify(genID)
-                        });
-      expect(badid.statusCode).toEqual(401);
-      expect(game.wordToGuess).toEqual('debug');
+  it('should be able to set a word', async (done) => {
+    callCount = 0;
+    socket.emit('setWord', 'debug');
+    setTimeout(() => {
+      expect(callCount).toEqual(1);
+      expect(result).toEqual({
+        display: ['_', '_', '_', '_', '_'],
+        isOver: false,
+        remainingGuesses: 6,
+        attempts: [],
+        isGameReady: false
+      })
       done();
-    });
+    }, 80)
```

Comment:
Does this set timeout just have to be higher than the previous it-block's timeout?

Reply:
Yep! I tried testing intervals of 5ms and 80 was the lowest amount of time it took to reliably pass the test I guess it's ~30ms more complex to compute on my computer for some reason..